lesson_06/expression/src/expression_calculator.ts

```diff
@@ -1,11 +1,29 @@
 export class ExpressionCalculator {
   /** Returns the calculation of ((a + b) * c) / d^e */
   calculate(a: number, b: number, c: number, d: number, e: number): number {
-    // Implement your code here to return the correct value.
-    return 0;
+    const P = this.add(a, b);
```

Comment:
In most cases, single character variable names are an anti-pattern and very hard to understand. These should be fully spelled out (e.g. `sum`, `product`, `exponent`, etc).